nautobot/virtualization/tables.py

```diff
@@ -23,14 +23,11 @@
     "VMInterfaceTable",
 )
 
-# TODO: re-introduce assign ip address button?
 VMINTERFACE_BUTTONS = """
-{% if perms.ipam.add_ipaddress %}
-    <!--
-    <a href="{% url 'ipam:ipaddress_add' %}?vminterface={{ record.pk }}&return_url={{ virtualmachine.get_absolute_url }}" class="btn btn-xs btn-success" title="Add IP address">
+{% if perms.ipam.add_ipaddress and perms.dcim.change_vminterface %}
+    <a href="{% url 'ipam:ipaddress_add' %}?vminterface={{ record.pk }}&return_url={% url 'virtualization:virtualmachine' pk=object.pk %}" class="btn btn-xs btn-success" title="Add IP address">
```

Comment:
Looks like the Interface button equivalent uses:

```suggestion
    <a href="{% url 'ipam:ipaddress_add' %}?vminterface={{ record.pk }}&return_url={{ request.path }}" class="btn btn-xs btn-success" title="Add IP address">
```

Is there a reason for not following that pattern?